init_update.lua

```diff
@@ -0,0 +1,183 @@
+local uv = vim.loop
+
+assert(vim.fn.executable "curl", "curl is not installed")
+assert(vim.fn.executable "unzip", "unzip is not installed")
+
+local init_path = debug.getinfo(1, "S").source:sub(2)
+local base_dir = init_path:match("(.*[/\\])"):sub(1, -2)
+
+if not vim.tbl_contains(vim.opt.rtp:get(), base_dir) then
+  vim.opt.rtp:append(base_dir)
+end
+
+require "lvim.utils.headless_fix"
+
+local bootstrap = require("lvim.bootstrap"):init(base_dir)
+
+local packer = require "packer"
+-- setup packer specifically for updating/installing
+packer.init {
+  auto_reload_compiled = false,
+  auto_clean = false,
+  log = {
+    highlights = false,
+    use_file = false,
+  },
+}
+
+local utils = require "lvim.utils"
+if not utils.is_directory(bootstrap.core_install_dir) then
+  assert(vim.fn.mkdir(bootstrap.core_install_dir), "unable to create core plugin install directory")
+end
+
+local function mkdir_tmp()
+  local path_sep = uv.os_uname().version:match "Windows" and "\\" or "/"
+  local tmpFilePath = os.tmpname()
+  vim.fn.delete(tmpFilePath)
+
+  local sep_idx = tmpFilePath:reverse():find(path_sep)
+  local path = tmpFilePath:sub(1, #tmpFilePath - sep_idx)
+  uv.fs_mkdtemp(path .. path_sep .. "lvim_core_dl_XXXXXX")
+  return path
+end
+
+local download_dir = mkdir_tmp()
+assert(download_dir, "unable to create core plugin download directory")
+
+local function get_lvim_after_user_config()
+  local original_lvim = lvim
+  local user_lvim = vim.deepcopy(lvim)
+  local original_package_loaded = package.loaded
+  local user_package_loaded = {}
+  _G.lvim = user_lvim
+  _G.package.loaded = user_package_loaded
+  local ok, err = pcall(dofile, require("lvim.config"):get_user_config_path())
+  if not ok then
+    print(err)
+  end
+  _G.lvim = original_lvim
+  _G.package.loaded = original_package_loaded
+
+  return user_lvim
+end
+
+local core_plugins = require "lvim.plugins"
+
+-- local d = require "deferred"
+-- Packer should now be configured and bootstrapped
+local a = require "packer.async"
+local async = a.sync
+local await = a.wait
+local wrap = a.wrap
+local jobs = require "packer.jobs"
+local result = require "packer.result"
+local async_stat = wrap(uv.fs_stat)
+
+---downloads and installs from a core plugin entry
+---@param plug table plugin entry
+---@return function
+local function download_and_install(plug)
```

Comment:
Why can't we write this in bash? I thought the whole point is to do this without relying on packer.

Reply:
> Why can't we write this in bash?

I'm not sure I understand. Why should this be written in bash? What's the benefit over it being written in Lua?

> I thought the whole point is to do this without relying on packer.

I'd like to clarify, this is to fix the _fragile behavior_ of Packer when syncing the core plugins using the built-in git file type handler. This is to avoid volatile state issues as I mentioned in my other comment.
Specifically, this replaces the _download_ and _install_, not the load step of Packer.